### ISSUE TITLE
KYC process has tons of bugs

### ISSUE THREAD

**Issue by MaxwellDG** [GaloyMoney/galoy-mobile#2985]
**Issue**: In the production version of the app, I tried to upgrade my account to level 2. This requires going through KYC. I'm only at the start and I've had 3 major errors:

- Making a 2nd attempt to type my name (after previously getting past the name page) resulted in being navigated back to the landing screen
- Trying to take a photo of my driver's licence front-side crashed the app once
- Trying to take a photo of my driver's licence front-side usually returns me to the 'Choose document' screen, and doesn't save any of my progress

**Solution**
@nicolasburtey @sandipndev I think it'd be best to discuss these in chat

**Comment by nicolasburtey** [comment]
yup I know thanks for pointing it out. I intend to spend more time on this to try to fix those issues. 

here is a useful pointer: https://developers.onfido.com/guide/sdk-webview-guide

note: we dont use a webview but rather an InAppBrowser as of now, but I'm thinking just using the Linking API might work better for permission related stuff

**Comment by MaxwellDG** [comment]
> yup I know thanks for pointing it out. I intend to spend more time on this to try to fix those issues in the next week or 2.
> 
> here is a useful pointer: https://developers.onfido.com/guide/sdk-webview-guide
> 
> note: we dont use a webview but rather an InAppBrowser as of now, but I'm thinking just using the Linking API might work better for permission related stuff

K glad you're aware of it. We'll discuss specifics on it later then. I read about what the best option is in the meantime

**Comment by jm9151821** [comment]
**Issue:** In the production version of the app, I tried to upgrade my account to level 2 using a iPhone device. I'm testing the KYC Onfido onboarding. After submitting the front and back of my ID and after the liveness check, the screen did not proceed further than just stuck loading.

Refreshed and it took me back to starting the whole process. To an user this can provide uneasiness as to where in the process he or she is and left. More odd behavior: I switched between applications after some 5 minutes and when I reverted back to the inapp webrowser inside blink, the onfido completion screen now shows. 
![IMG_8948 2](https://github.com/GaloyMoney/galoy-mobile/assets/32444279/557f514f-a4b9-435a-87b2-801951291e4c)

**Comment by jm9151821** [comment]
Customer using Huawei nova Nam lx9 - after inputting their name to proceed with KYC process, tapping on the button to 'Activate Camera' does not work in order to enable use of the camera and continue the process.

1st attempt within in-app browser
2nd attempt from normal phone browser. 

![WhatsApp Image 2024-02-22 at 3 35 28 PM](https://github.com/GaloyMoney/galoy-mobile/assets/32444279/2d3132ba-a926-44d5-892f-ec6d1dd17e3d)
![WhatsApp Image 2024-02-22 at 3 35 28 PM (1)](https://github.com/GaloyMoney/galoy-mobile/assets/32444279/35719db7-ce39-43fe-b4f9-40db0538f9bf)

Cc @nicolasburtey
